src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs

```diff
@@ -205,6 +199,16 @@ private static string GetDotnetPath()
             return new Muxer().MuxerPath;
         }
 
+        internal static Dictionary<string, string> GetMSBuildRequiredEnvironmentVariables()
+        {
+            return new()
```

Comment:
I know this is code that is already there but surprised this is using an ordinal comparison for environment variable names.

Reply:
Per POSIX spec, env vars with different casing should be treated as distinct:

> Environment variable names used by the utilities in the Shell and Utilities volume of POSIX.1-2017 consist solely of uppercase letters, digits, and the <underscore> ( '_' ) from the characters defined in [Portable Character Set](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap06.html#tagtcjh_3) and do not begin with a digit. Other characters may be permitted by an implementation; applications shall tolerate the presence of such names. Uppercase and lowercase letters shall retain their unique identities and shall not be folded together.